### PR TITLE
[client, proxy] Harden uspfilter conntrack and share TCP relay

### PR DESCRIPTION
**Pull request by lixmal**
## Describe your changes

Harden the userspace firewall and consolidate TCP byte-relaying.

- Fix TCP conntrack state-machine edge cases: retransmitted or wrong-direction segments no longer tear flows down prematurely.
- Tighten RST handling: ignore late RSTs in TimeWait and spoofed RSTs in SynSent; reject illegal flag combos.
- Add per-state TCP teardown timeouts with Linux-aligned defaults and env overrides.
- Cap conntrack table size with bounded-sample eviction and mobile-aware defaults, env-overridable.
- Extract a shared TCP relay helper with proper TCP half-close propagation; route the uspfilter forwarder, SSH forwarding, and the reverse-proxy relay through it (removing two duplicates).
- Guard hot-path log sites to eliminate argument allocations when the level is off, and fix a latent nil-argument formatting bug in the logger.
- Thread context through SSH local/remote port-forwarding so outer cancellation actually tears down in-flight relays.

### Checklist
- [x] Is it a bug fix
- [ ] Is a typo/documentation fix
- [x] Is a feature enhancement
- [x] It is a refactor
- [x] Created tests that fail without the change (if possible)

> By submitting this pull request, you confirm that you have read and agree to the terms of the [Contributor License Agreement](https://github.com/netbirdio/netbird/blob/main/CONTRIBUTOR_LICENSE_AGREEMENT.md).

## Documentation
Select exactly one:

- [ ] I added/updated documentation for this change
- [x] Documentation is **not needed** for this change (explain why)

Internal firewall and relay plumbing; no user-facing API changes. New NB_CONNTRACK_* tuning env vars are opt-in with safe defaults.

### Docs PR URL (required if "docs added" is checked)
Paste the PR link from https://github.com/netbirdio/docs here:

https://github.com/netbirdio/docs/pull/__

## Summary by CodeRabbit

* **New Features**
  * Configurable, platform-aware per-protocol conntrack caps; new robust bidirectional relay with context-aware cancellation, idle-timeouts, and half-close support.
* **Bug Fixes**
  * Hardened TCP state machine and RST/illegal-flag handling; eviction now prefers tombstoned entries to preserve live flows.
* **Performance / Logging**
  * Trace/error logs gated by logger level to reduce noise.
* **Tests**
  * Extensive tests for conntrack caps, TCP teardown/RST/state edge cases, and relay behavior.

### CODE REVIEW

File: proxy/internal/tcp/router.go [hunk 1]
```diff
@@ -528,11 +529,14 @@ func (r *Router) relayTCP(ctx context.Context, conn net.Conn, sni SNIHost, route
...
-		idleTimeout = DefaultIdleTimeout
+		idleTimeout = netrelay.DefaultIdleTimeout
 	}
 
 	start := time.Now()
-	s2d, d2s := Relay(svcCtx, entry, conn, backend, idleTimeout)
+	s2d, d2s := netrelay.Relay(svcCtx, conn, backend, netrelay.Options{
+		IdleTimeout: idleTimeout,
+		Logger:      entry,
+	})
```
[the remark this answers is not in the document]
Comment (lixmal) [reply proxy/internal/tcp/router.go:539]: Done

File: client/firewall/uspfilter/log/log.go [hunk 2]
```diff
@@ -209,16 +226,59 @@ func (l *Logger) Debug2(format string, arg1, arg2 any) {
...
+	msg := logMessage{level: level, argCount: uint8(n), format: format}
+	slots := [maxArgs]*any{&msg.arg1, &msg.arg2, &msg.arg3, &msg.arg4, &msg.arg5, &msg.arg6, &msg.arg7, &msg.arg8}
+	for i := 0; i < n; i++ {
+		*slots[i] = args[i]
+	}
+	select {
+	case l.msgChannel <- msg:
+	default:
+	}
+}
```
[the remark this answers is not in the document]
Comment (lixmal) [reply client/firewall/uspfilter/log/log.go:276]: Done

File: util/netrelay/relay.go [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+		case <-t.C:
+			last := time.Unix(0, lastActivity.Load())
+			if time.Since(last) >= idle {
+				idleHit.Store(true)
+				cancel()
+				return
+			}
+		}
+	}
+}
```
[the remark this answers is not in the document]
Comment (lixmal) [reply util/netrelay/relay.go:176]: Done

[the remark this answers is not in the document]
Comment (lixmal) [reply util/netrelay/relay.go:176]: Done

File: util/netrelay/relay.go [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+// returned because a relay always terminates on some kind of EOF/cancel.
+func Relay(ctx context.Context, a, b io.ReadWriteCloser, opts Options) (aToB, bToA int64) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	go func() {
+		<-ctx.Done()
+		_ = a.Close()
+		_ = b.Close()
+	}()
```
[the remark this answers is not in the document]
Comment (lixmal) [reply util/netrelay/relay.go:88]: done

[the remark this answers is not in the document]
Comment (lixmal) [reply util/netrelay/relay.go:88]: Done

**Review by pappz — APPROVED** [review]
